core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -65,6 +86,152 @@ public double getMarketPrice(String currencyCode) throws ExecutionException, Int
+        };
+
+        final Optional<Offer> highestSellPriceOffer = sellOffers.stream()
+        .filter(o -> o.getPrice() != null)
+        .max(Comparator.comparingLong(o -> o.getPrice().getValue()));
+
+        if (highestSellPriceOffer.isPresent()) {
+            final Offer offer = highestSellPriceOffer.get();
+        }
+
+        final Optional<Offer> highestSellVolumeOffer = sellOffers.stream()
+                .filter(o -> o.getVolume() != null)
+                .max(Comparator.comparingLong(o -> o.getVolume().getValue()));
+
+        if (highestSellVolumeOffer.isPresent()) {
+            final Offer offer = highestSellVolumeOffer.get();
+        }
+
+        // buildChartAndTableEntries called
+        accumulatedAmount = 0;
+        
+        // Create buyer hashmap {key:price, value:count}, uses TreeMap to sort by key (asc)
+        TreeMap<Double,Double> sellTM = new TreeMap<Double,Double>();               
+        for(Offer offer: sellOffers){            
+            Price price = offer.getPrice();
+            if (price != null) {
+                double amount = (double) offer.getAmount().value / LongMath.pow(10, offer.getAmount().smallestUnitExponent());
+                accumulatedAmount += amount;
+                double priceAsDouble = (double) price.getValue() / LongMath.pow(10, price.smallestUnitExponent());
+                sellTM.put(priceAsDouble, accumulatedAmount);
+                System.out.println("sell price");
+                System.out.println(priceAsDouble);
+                System.out.println("sell accumulated amount");
+                System.out.println(accumulatedAmount);
+            }
+        };
+
+        // Create buyer hashmap {key:price, value:count}, uses TreeMap to sort by key (asc)
+        // TreeMap<Double,Double> buyTM = new TreeMap<Double,Double>();        
+        // buyOffers.forEach((offer) -> {            
+        //     Double price = offer.getPrice().getDoubleValue() / offer.getOfferPayload().getCoinAmount();//Price per XMR coin;
+        //     Long previousValue = buyTM.get(price) != null ? buyTM.get(price) : Long.valueOf(0);
+        //     buyTM.put(price, previousValue + 1);
+
+        //     System.out.println("Offer price");
+        //     System.out.println(offer.getPrice().getDoubleValue());
+        //     System.out.println("1 XMR price in ETH");
+        //     System.out.println(price);
+        // });
+        
+        // // Create seller hashmap {key:price, value:count}, uses TreeMap to sort by key (asc)
+        // TreeMap<Double,Long> sellTM = new TreeMap<Double,Double>();
+        // sellOffers.forEach((offer) -> {
+        //     Double price = offer.getPrice().getDoubleValue() / offer.getOfferPayload().getCoinAmount();//Price per XMR coin
+        //     Long previousValue = sellTM.get(price) != null ? sellTM.get(price) : Long.valueOf(0);
+        //     sellTM.put(price, previousValue + 1);
+        // });
+        
+        System.out.println("loops done");
```

Comment:
same as above, please remove println and commented out code **IF not needed**

Reply:
done thanks (see explanation above)